### PR TITLE
Feature/#148 allow taipy enterprise scheduler

### PR DESCRIPTION
**Pull request by jrobinAV**
@toan-quach this is just a draft PR. The tests are failing, and the code is not reviewed. But the purpose is just to share my ideas.

Based on your code and your PR, I propose to change the scheduler factory so it is able to instantiate a Scheduler from taipy-enterprise. This new taipy-enterprise.scheduler should inherit from taipy-core._Scheduler and should contain the job recovery code. 

What do you think?

### CODE REVIEW

**Review by toan-quach — APPROVED** [review]
lgtm :D

File: taipy/core/_scheduler/_scheduler_factory.py [hunk 1]
```diff
@@ -17,10 +18,16 @@
...
+    def _build_scheduler(cls) -> _AbstractScheduler:
         if Config.job_config._is_default_mode():
-            _Scheduler._recover_jobs()
-            return _Scheduler
-
-        package = f"taipy.{Config.job_config.mode}.scheduler"
-        return _load_fct(package, "Scheduler")
+            if util.find_spec("taipy.enterprise"):
+                # Mode standalone with enterprise version
+                package = "taipy.enterprise.core.scheduler"
```
Comment (toan-quach) [inline taipy/core/_scheduler/_scheduler_factory.py:25]: I think I mentioned on the new enterprise pr, I think it'd be more consistent if we have something like this: `taipy/enterprise/core/scheduler.py` will become `taipy/enterprise/core/scheduler/scheduler.py`

**Review by toan-quach — APPROVED** [review]

File: taipy/core/_scheduler/_scheduler_factory.py (hunk 1 above)
Comment (jrobinAV) [reply taipy/core/_scheduler/_scheduler_factory.py:25]: yep. Good to me!

File: taipy/core/_scheduler/_scheduler_factory.py [hunk 2]
```diff
@@ -17,10 +18,16 @@
...
+                return _load_fct(package, "Scheduler")()
+            else:
+                # Mode standalone with community version
+                return _Scheduler()
+        else:
+            # Mode airflow with enterprise version
+            package = f"taipy.{Config.job_config.mode}.scheduler"
+            return _load_fct(package, "Scheduler")()
```
Comment (florian-vuillemot) [inline taipy/core/_scheduler/_scheduler_factory.py:33]: https://github.com/Avaiga/taipy-core/blob/develop/taipy/core/config/job_config.py

Comment (toan-quach) [reply taipy/core/_scheduler/_scheduler_factory.py:33]: not sure I understand what you mean?

Comment (jrobinAV) [reply taipy/core/_scheduler/_scheduler_factory.py:33]: 🤣

Comment (jrobinAV) [reply taipy/core/_scheduler/_scheduler_factory.py:33]: He meant that we could re-use _MODE_TO_MODULE to get the package name from the mode.
```
_MODE_TO_MODULE: Dict[str, str] = {
    "airflow": "taipy.airflow",
}
```

Comment (toan-quach) [reply taipy/core/_scheduler/_scheduler_factory.py:33]: ahhh i see i see, we surely can!

**Pull request closed without merging**